src/common.h

```diff
@@ -585,16 +585,31 @@ LY_ERR ly_strcat(char **dest, const char *format, ...);
     }
 
 /**
- * @brief allocate and insert new item inro linked list.
+ * @brief Allocate and insert new item into linked list, return in case of error.
  *
- * @param[in] CTX used for loggin.
- * @param[in,out] LIST Linker list to add to.
- * @param[out] NEW_ITEM New item that was appended to the list.
- * @param[in] LINKER name of structuin member that is used to connect items together.
+ * @param[in] CTX used for logging.
+ * @param[in,out] LIST Linked list to add to.
+ * @param[out] NEW_ITEM New item that is appended to the list.
+ * @param[in] LINKER name of structure member that is used to connect items together.
  */
 #define LY_LIST_NEW_RET(CTX, LIST, NEW_ITEM, LINKER) \
```

Comment:
It is common actually to be able to set return code via RETVAL parameter to be able to use the macro even in functions returning pointers or somehow limited to certain return codes, see e.g. LY_ARRAY_NEW_RET